Source/CarthageKit/Xcode.swift

```diff
@@ -680,33 +679,56 @@ public func buildInDirectory(_ directoryURL: URL, withOptions options: BuildOpti
 					} else {
 						filteredSDKs = sdks.filter { options.platforms.contains($0.platform) }
 					}
-
 					return sdkFilter(filteredSDKs, scheme, configuration, project)
 				}
 
-				let buildProgress = buildScheme(scheme, withConfiguration: options.configuration, inProject: project, workingDirectoryURL: directoryURL, derivedDataPath: options.derivedDataPath, toolchain: options.toolchain, sdkFilter: wrappedSDKFilter)
+				return buildScheme(scheme, withOptions: options, inProject: project, workingDirectoryURL: directoryURL, sdkFilter: wrappedSDKFilter)
 					.mapError { (error) -> CarthageError in
 						if case let .taskError(taskError) = error {
 							return .buildFailed(taskError, log: nil)
 						} else {
 							return error
 						}
 					}
-					// Discard any existing Success values, since we want to
+					.on(started: {
+						observer.send(value: .success(initialValue))
+					})
+			}
+			.startWithSignal({ (signal, signalDisposable) in
+				disposable += signalDisposable
+
+				let ignoredValue = (ProjectLocator.workspace(URL(string: ".")!), "")
+				
+				let eventSignal: Signal<TaskEvent<(ProjectLocator, String)>, CarthageError> = signal
+					// Discard any Success values, since we want to
 					// use our initial value instead of waiting for
 					// completion.
-					.map { taskEvent in
-						return taskEvent.map { _ in initialValue }
+					.map { (taskEvent: TaskEvent<URL>) -> TaskEvent<(ProjectLocator, String)> in
+						return taskEvent.map { _ in ignoredValue }
 					}
-					.filter { taskEvent in taskEvent.value == nil }
-
-				return BuildSchemeProducer(value: .success(initialValue))
-					.concat(buildProgress)
-			}
-			.startWithSignal { signal, signalDisposable in
-				disposable += signalDisposable
-				signal.observe(observer)
-			}
+					.filter { taskEvent in
+						return taskEvent.value == nil
+					}
+				
+				let versionFileSignal: Signal<TaskEvent<(ProjectLocator, String)>, CarthageError> = signal
+						.ignoreTaskData()
+						.collect()
+						.flatMap(.concat) { (urls: [URL]) -> SignalProducer<(), CarthageError> in
+							guard let dependency = dependency, let rootDirectoryURL = rootDirectoryURL else {
+								return .empty
+							}
+							return createVersionFile(for: dependency, platforms: options.platforms, buildProducts: urls, rootDirectoryURL: rootDirectoryURL)
+								.flatMapError { _ in .empty }
+						}
+						.map { _ -> TaskEvent<(ProjectLocator, String)> in
+							return .success(ignoredValue)
+						}
+						.filter { taskEvent in
+							return false
+						}
+				
+				Signal.merge([eventSignal, versionFileSignal]).observe(observer)
```

Comment:
just to clarify a bit, this new collect extension would 'collect' on the 'success' task events, but let the rest through.

So instead of a stream of stdin, success(val1), stdin, succes(val2) we'd get 'stdin, stdin, success([val1, val2]) ?

should this extension just live in Xcode.swift for now?